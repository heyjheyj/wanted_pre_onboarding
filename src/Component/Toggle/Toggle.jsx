import React, { useState } from "react";
import styles from "./Toggle.module.css";

const Toggle = props => {
  const [isSwitchOn, setIsSwitchOn] = useState(false);

  const toggleSwitch = e => setIsSwitchOn(e.target.checked);

  return (
    <div className={styles.container}>
      <h3 className={styles.title}>Toggle</h3>
      <div className={styles.toggleSwitch}>
        <input
          type="checkbox"
          className={styles.checkbox}
          id="toggleSwitch"
          onClick={toggleSwitch}
          data-testid="test-togglecheckbox"
        />
        <label className={styles.label} htmlFor="toggleSwitch">
          <span className={styles.toggleInner} />
          <span className={styles.switch} />
        </label>
      </div>
      <span className={styles.text} data-testid="test-text">
        Toggle Switch {isSwitchOn ? "ON" : "OFF"}
      </span>
    </div>
  );
};

export default Toggle;
