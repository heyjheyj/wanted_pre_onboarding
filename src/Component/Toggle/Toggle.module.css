.container {
  width: 80%;
  height: 200px;
  align-items: center;
  border: 1px solid grey;
  justify-content: center;
  display: flex;
  flex-direction: column;
  position: relative;
  border-radius: 10px;
}

.title {
  position: absolute;
  top: 5px;
  left: 5px;
}

.toggleSwitch {
  position: relative;
  width: 75px;
  display: inline-block;
  text-align: left;
  top: 8px;
}

.checkbox {
  display: none;
}

.label {
  display: block;
  overflow: hidden;
  cursor: pointer;
  border: 0 solid #bbb;
  border-radius: 20px;
}

.toggleInner {
  display: block;
  width: 200%;
  margin-left: -100%;
  transition: margin-left 0.4s ease-in;
}

.toggleInner:before,
.toggleInner:after {
  float: left;
  width: 50%;
  height: 36px;
  padding: 0;
  box-sizing: border-box;
  content: "";
}

.toggleInner:before {
  background-color: #431cc6;
}

.toggleInner:after {
  background-color: #bbb;
}

.switch {
  display: block;
  width: 24px;
  margin: 5px;
  background: #fff;
  position: absolute;
  top: 0;
  bottom: 0;
  right: 40px;
  border: 0 solid #bbb;
  border-radius: 50%;
  transition: all 0.4s ease-in;
}

.checkbox:checked + .label .toggleInner {
  margin-left: 0;
}

.checkbox:checked + .label .switch {
  right: 0px;
}

.text {
  margin-top: 20px;
  color: gray;
  font-weight: 600;
}