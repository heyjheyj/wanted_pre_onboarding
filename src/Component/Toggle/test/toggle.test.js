import React from "react";
import TestRenderer from "react-test-renderer";
import Toggle from "../Toggle";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("Toggle", () => {
  it("renders", () => {
    const component = TestRenderer.create(<Toggle />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  describe("Toggle Switch", () => {
    let togglebox;
    let text;

    beforeEach(() => {
      render(<Toggle />);

      togglebox = screen.getByTestId("test-togglecheckbox");
      text = screen.getByTestId("test-text");
    });

    it("turns on Switch and changes text when toggle is clicked", () => {
      userEvent.click(togglebox);

      expect(togglebox.checked).toBe(true);
      expect(text.textContent).toBe("Toggle Switch ON");
    });

    it("turn off Switch and changes text when toggle is clicked", () => {
      userEvent.click(togglebox);
      userEvent.click(togglebox);

      expect(togglebox.checked).toBe(false);
      expect(text.textContent).toBe("Toggle Switch OFF");
    });
  });
});
