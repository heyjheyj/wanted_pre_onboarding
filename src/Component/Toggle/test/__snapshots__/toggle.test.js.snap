// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle renders 1`] = `
<div
  className="container"
>
  <h3
    className="title"
  >
    Toggle
  </h3>
  <div
    className="toggleSwitch"
  >
    <input
      className="checkbox"
      data-testid="test-togglecheckbox"
      id="toggleSwitch"
      onClick={[Function]}
      type="checkbox"
    />
    <label
      className="label"
      htmlFor="toggleSwitch"
    >
      <span
        className="toggleInner"
      />
      <span
        className="switch"
      />
    </label>
  </div>
  <span
    className="text"
    data-testid="test-text"
  >
    Toggle Switch 
    OFF
  </span>
</div>
`;
